internal/pkg/manifest/errors.go

```diff
@@ -19,14 +19,15 @@ func (e *ErrInvalidWorkloadType) Error() string {
 	return fmt.Sprintf("invalid manifest type: %s", e.Type)
 }
 
-// ErrInvalidPipelineManifestVersion occurs when the pipeline.yml file
+// ErrInvalidPipelineManifestVersion occurs when the pipeline.yml/manifest.yml file
 // contains invalid schema version during unmarshalling.
 type ErrInvalidPipelineManifestVersion struct {
 	invalidVersion PipelineSchemaMajorVersion
+	fileName       string
 }
 
 func (e *ErrInvalidPipelineManifestVersion) Error() string {
-	return fmt.Sprintf("manifest.yml contains invalid schema version: %d", e.invalidVersion)
+	return fmt.Sprintf("%s contains invalid schema version: %d", e.fileName, e.invalidVersion)
```

Comment:
Alternatively, what do you think of not including path name in errors thrown by `manifest` package?

```
func (e *ErrInvalidPipelineManifestVersion) Error() string {
	return fmt.Sprintf("pipeline manifest contains invalid schema version: %d", e.invalidVersion)
```

and then in `workspace` package:
```
pipelineManifest, err := manifest.UnmarshalPipeline(data)
	if err != nil {
		return nil, fmt.Errorf("unmarshal pipeline manifest %s: %w", path, err)
	}
```

I feel like `manifest` package should be agnostic of `path` information, while it makes sense for `workspace` to know `path`.

Let me know if this feels right to you!

Reply:
Sure. We can separate concerns. I guess it's already clear enough which file "pipeline manifest" refers to.